src/Entity/IdeasWorkshop/Need.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace AppBundle\Entity\IdeasWorkshop;
+
+use Algolia\AlgoliaSearchBundle\Mapping\Annotation as Algolia;
+use AppBundle\Entity\EntityEnabledTrait;
+use AppBundle\Entity\EntityNameSlugTrait;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(
+ *     uniqueConstraints={
+ *         @ORM\UniqueConstraint(name="need_slug_unique", columns="slug")
+ *     }
+ * )
+ *
+ * @UniqueEntity("name")
```

Comment:
why do you not use slug property here?

Reply:
added constraint and unique entity on both slug & name